spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -836,19 +836,47 @@ public static String quotedFullIdentifier(String catalogName, Identifier identif
    * @param format format of the file
    * @param partitionFilter partitionFilter of the file
    * @return all table's partitions
+   * @deprecated use {@link Spark3Util#getPartitions(SparkSession, Path, String, Map,
+   *     PartitionSpec)}
    */
+  @Deprecated
```

Comment:
Nit: i think its still valid to have this API, and no need to deprecate?  If you want to infer the partition spec from the path.

Reply:
now that the other method spec is not optional, I think we can remove deprecated flag for this method.
However I feel that as we know spark interpretation doesn't work for all cases and users will always have a partition spec, it might be better to deprecate this as `add_files` doesn't validate schema and does not fail fast. if a partition type is not interpreted correctly add_files doesn't fail and users will end up with bad data in their table.